sailfishusermanager.cpp

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright (c) 2020 Jolla Ltd.
+ * Copyright (c) 2020 Open Mobile Platform LLC.
+ *
+ * All rights reserverd.
+ *
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include "sailfishusermanager.h"
+#include "usermanager_adaptor.h"
+
+#include <QDBusConnection>
+#include <QTimer>
+#include <QFile>
+#include <QDir>
+
+#include <pwd.h>
+#include <grp.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <libuser/user.h>
+
+const char* DBUS_INTERFACE = "org.sailfishos.usermanager";
+const char* DBUS_OBJECT_PATH = "/";
```

Comment:
While we have quite a few components that have path /, it should be in the format /org/sailfishos/usermanager although it might sound a bit redundant. One reasoning here http://0pointer.de/blog/projects/versioning-dbus.html.

Reply:
The referenced page doesn't actually give any reasoning, it just says that it's a bad idea because "it makes it very hard to implement more than one service or service version on a single D-Bus connection" but doesn't explain why. I never noticed any difficulties with `/` as a path. AFAICT it's as easy and simple as using any other path.

Reply:
Technically, yes, path is a path and does not matter. 

However, given that there is documented convention, following the convention is less "surprising" than deviating from it -> I'd reverse the question: What if any benefit there is from using "/"?

Reply:
Well, my reasoning behind '/' is that since the path has no meaning for this interface that's the shortest one that works. I have no preference in this so whatever works for me.

Reply:
For a more authoritative source for prefixing (or namepacing) object paths, it is recommended by FDo's [tutorial](https://dbus.freedesktop.org/doc/dbus-tutorial.html#objects).

Reply:
That [tutorial](https://dbus.freedesktop.org/doc/dbus-tutorial.html#objects) still leaves unanswered the question - if interface is already prefixed then why add the same prefix to the object path? It just says "Namespacing object paths is smart, by starting them with the components of a domain name you own". O'rly? My guess it's that just somebody likes it that way. Then I would leave it up to the author here too.

Reply:
I vote "up to author" too.